File: src/snippets/01-packet.sn.cpp
#include <ndn-cxx/face.hpp>

int main(int argc, char** argv)
{
    // Create an NDN Name from a URI string
    ndn::Name name("/edu/ucla/cs/135/notes");

    // Create an Interest packet with this name
    ndn::Interest interest(name);

    // Set the Interest packet's InterestLifetime to 4 seconds (default)
    interest.setInterestLifetime(ndn::time::seconds(4));

    // Create a Data packet with the same name
    ndn::Data data(name);

    // Set the Data packet's content to "Hello, World!"
    data.setContent(reinterpret_cast<const uint8_t*>("Hello, World!"), 13);
}

File: src/snippets/01-packet.sn.py
from ndn.encoding import Name
from ndn.types import Interest

# Create an NDN name from a URI string
name = Name.from_str('/edu/ucla/cs/135/notes')

# Create an Interest packet with this name
interest = Interest(name)

# Set the Interest packet's InterestLifetime to 4 seconds (default)
interest.interest_lifetime = 4000
